src/main/java/hudson/plugins/octopusdeploy/OctopusDeployPlugin.java

```diff
@@ -193,6 +192,27 @@ public FormValidation doCheckApiKey(@QueryParameter String apiKey) {
             return FormValidation.ok();
         }
 
+        private static boolean isApiKeyValid(String apiKeyValue) {
+            final String apiKeyPrefix = "API-";
+            if (!apiKeyValue.startsWith(apiKeyPrefix)) {
+                return false;
+            }
+
+            int keyLength = apiKeyValue.length();
+            if (keyLength < 29 || keyLength > 36) {
```

Comment:
As @benPearce1 pointed out, we still need to allow API key less than 32 for backward compatibility although the new minimum length for API key is 32.